src/xrc/src/api.rs

```diff
@@ -370,7 +374,9 @@ async fn handle_crypto_base_fiat_quote_pair(
     }
 
     let base_asset = base_asset.clone();
-    with_request_counter(num_rates_needed, async move {
+    let available_exchanges_count = EXCHANGES.iter().filter(|e| e.is_available()).count();
+    let http_requests_needed = num_rates_needed.saturating_mul(available_exchanges_count);
```

Comment:
So now we have these two lines three times outside of the guard as opposed to twice inside the guard (in the functions `new` and `drop`). Is there still a net gain...?

Reply:
I do not think there is a gain. I see it more as a loss at this point (up to the programmer to supply the request amount). I am going to move it back into the guard as that seems safer long term.